src/main/scala/dsls/regex/RegularExpression.scala

```diff
@@ -11,6 +11,30 @@ package dsls.regex
 abstract class RegularExpression {
   /** returns true if the given string matches this regular expression */
   def matches(string: String) = RegexMatcher.matches(string, this)
+  def ||(right: RegularExpression): Union =
+    Union(this,right)
+  def ~(right: RegularExpression): Concat =
+    Concat(this,right)
+  def <*>(): Star =
+    Star(this)
+  def <+>(): Concat =
+    Concat(this, Star(this))
+  def repeat(n: Int): Concat = {
+    /** I don't feel that this is as functional as it should be and
+     *  I would love some feedback on how to make it better stylistically
+     *  I'm also not sure how to get the desired {3} syntax so instead have
+     *  implemented it as repeat(3)
+     */
+    if (n == 1)
+      Concat(this,null)
+    if (n ==2)
+      Concat(this,this)
+    val temp = this
+    var retVal: Concat = Concat(this,this)
+    for (i <- 3 to n) { retVal = Concat(retVal,temp) }
+    retVal
+  }
+  
```

Comment:
I just used pure recursion: the base case, where n is 0 returned `EPSILON`, and otherwise it returned `this ~ this(n-1)`.  I think your solution is very interesting since it involves the more imperative parts of Scala like for loops and vars that I'm not yet familiar with.  Also, `repeat` was a very intuitive choice of function name--perhaps even clearer than just `{n}`--thought it might end up being a bit verbose.